walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,93 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.io.*;
+import java.util.*;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    long duration = request.getDuration();
+    Collection<TimeRange> ranges = new ArrayList<>();
+
+    //--------------------------------------------------------------------edge cases
+    // if duration too long
+    if (duration > TimeRange.WHOLE_DAY.duration()) {
+      return ranges;
+    }
+
+    // if no attendees
+    if (attendees.isEmpty()) {
+      ranges.add(TimeRange.WHOLE_DAY);
+      return ranges;
+    }
+
+    //----------------------------------------------------------------- general case
+    if (events.size() >= 0) {
+      // iterate through each element in collection
+      Iterator<Event> itr = events.iterator();
+      ArrayList<ArrayList<Integer>> meetings = new ArrayList<ArrayList<Integer>>();
```

Comment:
Generally preferable to use the interface on the left:
```suggestion
      List<List<Integer>> meetings = new ArrayList<ArrayList<>>();
```